Lib/test/test_threading.py

```diff
@@ -1106,6 +1106,26 @@ def exit_handler():
         self.assertEqual(out, b'')
         self.assertIn(b"can't create new thread at interpreter shutdown", err)
 
+    def test_start_new_thread_failed(self):
+        # gh-109746: if Python fails to start newly created thread
+        # due to failure of underlying PyThread_start_new_thread() call,
+        # its state should be removed from interpreter' thread states list
+        # to avoid its double cleanup
+        code = """if 1:
+            import resource
+            import threading
+
+            resource.setrlimit(resource.RLIMIT_NPROC, (150, 150))
```

Comment:
Should not it be `0, 0`?

It looks to me, that it is a total number of threads/processes. It is not increased in the loop, because a new thread is joined. If first thread was successfully created, all following threads will be successfully created as well.